### DISCUSSION TITLE
What happens to old connection upon its update with no explicit close or shutdown

### DISCUSSION THREAD

**Discussion by Unniboy** [redis/lettuce#1936] [Q&A]
Lets say that during application boot up, I start a cluster connection
```
@Getter
private StatefulRedisClusterConnection<String, String> connection;

public void connectToRedis(String connectionInfo) {

    // Syntax: redis://[password@]host[:port]
    // Syntax: redis://[username:password@]host[:port]
    RedisClusterClient redisClient = RedisClusterClient.create(connectionInfo);

    connection = redisClient.connect();

    //connection is used across threads to do redis operations.......
}
```

Later at some point, I need to update the `connection` as the redis node has moved/changed.
- Can I simply invoke `connectToRedis` with new `connectionInfo`? If yes, what happens to the `connection` currently being used?
- Do I need to `close`/`shutdown` `connection`/`redisClient` respectively before using the new `connectionInfo`?

**Comment by mp911de** [comment]
> What happens to old connection upon its update with no explicit close or shutdown

They remain open. Each time you call `connect()`, a new connection is opened.
